src/directive.js

```diff
@@ -6,8 +6,12 @@ export default {
     el = core.getInputElement(el)
     el.addEventListener('input', core.inputHandler, true)
 
+    const config = core.normalizeConfig(binding.value)
+
+    if (binding.modifiers && binding.modifiers.short) config.short = true
```

Comment:
The modifier is only being applied during bind, so it will be wiped away in the update hook if the mask gets changed dynamically.   A better approach would be to pass the modifiers as input to normalizeConfig in both bind and update then store in there when we are building the config object.